hugr-py/src/hugr/tys.py

```diff
@@ -82,6 +91,10 @@ def resolve(self, registry: ext.ExtensionRegistry) -> Type:
         """Resolve types in the type using the given registry."""
         return self
 
+    def to_model(self) -> model.Term | model.Splice:
+        """Convert the type to a model Term."""
+        raise NotImplementedError(self)
```

Comment:
The absence of implementations for `Type`, `TypeParam`, `TypeArg` seem like significant omissions! Is there a reasoning behind it?

Reply:
These classes are protocols. The actual types, type params, etc inherit from the class and implement the method accordingly.